Build/validate-qir.ps1

```diff
@@ -89,35 +104,71 @@ $qirProjects = @(
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics EvaluatingFunctions EvaluatingFunctions.csproj); Args = @() },
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics ResourceCounting ResourceCounting.csproj); Args = @() },
 
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation h2 command-line H2SimulationSampleCmdLine.csproj); Args = @() },
-    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising adiabatic AdiabaticIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising generators IsingGeneratorsSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising phase-estimation IsingPhaseEstimationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising simple SimpleIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising trotter-evolution IsingTrotterSample.csproj); Args = @() },
-    # needs argument(s) #@{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @("--numTrials", "20") }
 )
 
+Write-Host "##[info]Beginning Microsoft.Quantum.Qir.CommandLineTool installation."
 if (-not (Get-Command qir-cli -ErrorAction SilentlyContinue)) {
-    Write-Host "##[error]The qir-cli command is missing; you can install it by running `dotnet tool install Microsoft.Quantum.Qir.CommandLineTool -g`.";
-    $script:allOk = $False
+    $version = ($Env:NUGET_VERSION -split "-")[0] + "-alpha"
+    Write-Host "##[info]Going to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+    dotnet tool install Microsoft.Quantum.Qir.CommandLineTool --version $version -g
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##[error]Failed to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+        $script:allOk = $False
+    } else {
+        Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool.$version has been installed."
+    }
 } else {
+    Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool is already installed."
+}
+
+if ($allOk) {
+    if ($IsWindows) {
+        $old_PATH = $env:PATH
+        if (!(Get-Command clang -ErrorAction SilentlyContinue)) {
+            if ((choco find --idonly -l llvm) -notcontains "llvm") {
+                Write-Host "##[info]Installing LLVM with Chocolatey."
+                choco install llvm --version=11.1.0
+            }
+
+            if ((choco find --idonly -l llvm) -contains "llvm") {
```

Comment:
This runs `choco find --idonly -l llvm` twice... as a possible optimization, you may want to change this to an `else` block.

Reply:
This is intentional. The first time is to check if it should do the install. The second is to check that the install was successful, and if so, proceed with the validation.

Reply:
A comment indicating that may be helpful, then. As written, it's not clear that the second call is to check for errors.

Reply:
added